packages/@aws-cdk/aws-codedeploy/lib/ecs/deployment-group.ts

```diff
@@ -55,8 +212,135 @@ export class EcsDeploymentGroup {
     return new ImportedEcsDeploymentGroup(scope, id, attrs);
   }
 
-  private constructor() {
-    // nothing to do until CFN supports ECS deployment groups
+  public readonly application: IEcsApplication;
+  public readonly deploymentGroupName: string;
+  public readonly deploymentGroupArn: string;
+  public readonly deploymentConfig: IEcsDeploymentConfig;
+  /**
+   * The service Role of this Deployment Group.
+   */
+  public readonly role: iam.IRole;
+
+  private readonly alarms: cloudwatch.IAlarm[];
+
+  constructor(scope: Construct, id: string, props: EcsDeploymentGroupProps) {
+    super(scope, id, {
+      physicalName: props.deploymentGroupName,
+    });
+
+    this.application = props.application || new EcsApplication(this, 'Application');
+    this.alarms = props.alarms || [];
+
+    this.role = props.role || new iam.Role(this, 'ServiceRole', {
+      assumedBy: new iam.ServicePrincipal('codedeploy.amazonaws.com'),
+    });
+
+    this.role.addManagedPolicy(iam.ManagedPolicy.fromAwsManagedPolicyName('AWSCodeDeployRoleForECS'));
+    this.deploymentConfig = props.deploymentConfig || EcsDeploymentConfig.ALL_AT_ONCE;
+
+    for (const service of props.services) {
```

Comment:
After thinking about this more, the relationship between ECS and CodeDeploy feels like an [integration](https://github.com/aws/aws-cdk/blob/v1-main/docs/DESIGN_GUIDELINES.md#integrations) to me.

We might want a new `aws-ecs-deployments` package that has a `CodeDeployDeployment` (or something) construct.

```ts
const service = new ecs.Service(this, 'Service', {...});
service.addDeployment(new ecs_deployments.CodeDeployDeployment({...}));
```

Reply:
Feels like that would add more inconsistency at this point -- We already have `ServerDeploymentGroup` and `LambdaDeploymentGroup` in the codedeploy package, plus the `EcsDeploymentGroup` class is referred to in `CodeDeployEcsDeployAction` in the codepipeline package.

Reply:
>  but all except CodeDeploy deployments aren't integrations with other services (they are configurations on the ECS service, built-in to ECS)

There is also the `CfnCodeDeployBlueGreenHook` and in the future there could be others. 

> Feels like that would add more inconsistency at this point -- We already have ServerDeploymentGroup and LambdaDeploymentGroup in the codedeploy package, plus the EcsDeploymentGroup class is referred to in CodeDeployEcsDeployAction in the codepipeline package.

I do think the EcsDeploymentGroup construct should exist here, but I think we need an integration construct that handles integrating this with the ECS service. I think the reason we need an integration library is because the ideal implementation leads to a circular dependency. Below line `241` we are editing the properties on an ECS service out of band, which we want to avoid.


```ts
const service = new ecs.Service(this, 'Service', {...});
service.addDeployment(new ecs_deployments.CodeDeployDeployment({...}));

// ecs service constructor
const deploymentConfig = this.deployment.bind(this);
new CfnService(this, 'Service', {
  deploymentController: deploymentConfig.deploymentController,
});
```

Reply:
@clareliguori I like @rix0rrr's suggestion which would remove the need for an integration library.

Reply:
Got it, makes sense.  I'd want to involve the ECS team in designing a new ecs deployments package.  Is it ok if we keep this PR scoped to the CodeDeploy EcsDeploymentGroup construct, and tackle the ECS-CodeDeploy integration package in a separate PR?  I can remove all the logic in this PR around editing the ECS service, so the construct stands a bit more on its own.